### PR TITLE
fix(make): enabling debug build

### PR DESCRIPTION
**Pull request by Alexandr-Solovev**
## Description

This PR introduces a fix for building oneDAL with REQDBG option. The correct opts have been added and aligned with `-DMKL_ILP64` and `-Zl`. Also pedantic opts for icx and dpcpp compiler have been updated. The `fno-system-debug` option has been added for the oneDAL build size optimization.
Lists of pedantic flags have been updated and aligned for `icx.mkl.32e.mkl` and `dpcpp.mk`.
Incorrect initialization of `dtrees` structures has been fixed.
New flags `-ZL_DPCPP` and `-DEBC_DPCPP` have been introduced for avoiding usage of `COMPILER` flags in compiling and linking dpcpp part.

---

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.
- [x] I have provided justification why quality metrics have changed or why changes are not expected.
- [x] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

**Review by ethanglaser — APPROVED** [review]

File: makefile [hunk 1]
```diff
@@ -116,26 +116,27 @@ AR_is_$(subst $(space),_,$(origin AR)) := yes
...
--DEBJ  := $(if $(REQDBG),-g,-g:none)
--DEBL  := $(if $(REQDBG),$(if $(OS_is_win),-debug,))
--EHsc  := $(if $(OS_is_win),-EHsc,)
--isystem := $(if $(OS_is_win),-I,-isystem)
--sGRP  = $(if $(OS_is_lnx),-Wl$(comma)--start-group,)
--eGRP  = $(if $(OS_is_lnx),-Wl$(comma)--end-group,)
+-Zl         := $(-Zl.$(COMPILER))
+-Zl_DPCPP   := $(-Zl.dpcpp)
+-DEBC       := $(if $(REQDBG),$(-DEBC.$(COMPILER)) -DDEBUG_ASSERT -DONEDAL_ENABLE_ASSERT) -DTBB_SUPPRESS_DEPRECATED_MESSAGES -D__TBB_LEGACY_MODE
+-DEBC_DPCPP := $(if $(REQDBG),$(-DEBC.dpcpp) -DDEBUG_ASSERT -DONEDAL_ENABLE_ASSERT) -DTBB_SUPPRESS_DEPRECATED_MESSAGES -D__TBB_LEGACY_MODE
```
Comment (Vika-F) [inline makefile:134]: Not sure that `-DTBB_SUPPRESS_DEPRECATED_MESSAGES -D__TBB_LEGACY_MODE` are needed here.
Please check.
Other parts of the PR look good to me.

**Pull request merged**